### PR TITLE
Upgrade to PHPStan 1.0

### PR DESCRIPTION
**Pull request by TomasVotruba**
- [#22]: upgrade phpstan to 1.0
- [#22]: use PHPStanTestCase super type
- [#22]: remove deprecated method
- [#22]: use interface in test property annotation
- [#22]: use self defined super type for test cases
- [#22]: ignore missed class for linter action
- [#22]: ignore PHPStanTestCase issue with config
- remove unused AstractTestCase
- fixup! remove unused AstractTestCase

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -25,18 +25,7 @@ jobs:
...
-        with:
-          path: vendor
-          key: ${{ runner.os }}-php-${{ matrix.php }}-composer-${{ hashFiles('**/composer.json') }}-${{ matrix.prefer }}-
-          restore-keys: |
-            ${{ runner.os }}-php-${{ matrix.php }}-composer-${{ matrix.prefer }}-
-
-      - name: Install dependencies
-        if: steps.composer-cache.outputs.cache-hit != 'true'
-        run: composer update --prefer-${{ matrix.prefer }} --prefer-dist --no-progress
+      - uses: "ramsey/composer-install@v1"
```
Comment (TomasVotruba) [inline .github/workflows/ci.yml:28]: This package handles composer cache out of the box: https://github.com/ramsey/composer-install#about

**Comment by TomasVotruba** [comment]
Ready for review :+1:  :tada:

**Comment by TomasVotruba** [comment]
Hm, it seems abandoned :cry: 

I'm currenctly trying another extension, from the same vendor as PHP Enum package: https://github.com/marc-mabe/php-enum-phpstan

**Comment by TomasVotruba** [comment]
I can confirm it works great :heavy_check_mark:

**Comment by simPod** [comment]
@TomasVotruba I tried using the another extension it and then noticed it's for another enum library. I'll rather fork this and switch to PHP 8.1 in several days.

**Comment by TomasVotruba** [comment]
@simPod Same here, the docs is very poor in both of them and requires lot of duplicated docblocks.
Can't wait for PHP 8.1 :).

If you work, let me know, I'd use it. I've burned many hours on this without any result nor feedback so I'm discouraged to do more.

**Comment by simPod** [comment]
I ended up with `composer require --dev timeweb/phpstan-enum:"dev-22-upgrade-phpstan-to-1.0"` (using lockfile)

Will do until php 8.1, I don't see any benefit of dedicating more time to it. :P

**Pull request closed without merging**